app/assets/javascripts/darkswarm/services/checkout.js.coffee

```diff
@@ -122,3 +124,38 @@ angular.module('Darkswarm').factory 'Checkout', ($injector, CurrentOrder, Shippi
     terms_and_conditions_accepted: ->
       terms_and_conditions_checkbox = angular.element("#accept_terms")[0]
       terms_and_conditions_checkbox? && terms_and_conditions_checkbox.checked
+    confirmExit: ->
+      if not @orderPlaced()
+        @intendedPage = $location.url()  # Redirect here if chosen to
+        modalInstance = $uibModal.open(
+          animation: true
+          template: '''
+            <div class="modal-header">
+              <h4 class="modal-title">Confirmation</h4>
+            </div>
+            <div class="modal-body">
+              Are you sure you want to leave this page? Your order has not been placed.
+            </div>
+            <div class="modal-footer">
+              <button class="btn btn-primary" ng-click="$close(true)">Yes</button>
+              <button class="btn btn-secondary" ng-click="$dismiss()">No</button>
```

Comment:
These messages will all need to be translatable for other languages.

Reply:
I'm not sure, but I think in this case, it would be best to create this as a hidden template in the checkout page, which I guess can be loaded like `template: document.getElementById("confirm_exit").innerHTML`.

I would try adding it inside a `<template id="confirm_exit">` tag in the [summary view](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/views/split_checkout/_summary.html.haml). You'll see examples of translated strings there, which can be found in [en.yml](https://github.com/openfoodfoundation/openfoodnetwork/blob/cb5164ac6e624756d52915194f49dde2a8dad6c1/config/locales/en.yml#L2179).

For more info about translations see https://github.com/openfoodfoundation/openfoodnetwork/wiki/Internationalisation-(i18n).

Let me know if you're able to go ahead with this, or if you have a better idea.

Reply:
Note that you only need to add the English text in en.yml. After the PR is merged, translators will be able to create the relevant translations for other languages.

Reply:
Okay, got it, thank you!